backend/src/backend/databse/__init__.py

```diff
@@ -33,24 +33,11 @@ def my_loads(s, *args, **kwargs):
     )
 
 
-def dbsession(func):
-    """Decorator to create an SQLAlchemy ORM session object and wrap the function
-    inside the session. A `session` argument is automatically set. Commit is
-    automatically performed when the function finish (and before returning to
-    the caller). Should any exception arise, rollback of the transaction is also
-    automatic.
+def get_session() -> OrmSession:
```

Comment:
Signature is incorrect ; this is a Generator. I thought it was a mistake on your end but turning this into a function just silently fails to record to DB!!
Name should thus also be more appropriate ; like `gen_session()`

Reply:
I don't quite understand here. Does it mean that I need to do the `commit` and `rollback` in this part?
like this?
```python
def gen_session() -> Generator[OrmSession, None, None]:
    with Session.begin() as session:
        try:
            yield session
            session.commit()
        except Exception as e:
            logger.info(e)
            session.rollback()
        finally:
            session.close()

```